src/main/java/ma/norsys/bookstore/services/BookServiceImp.java

```diff
@@ -0,0 +1,89 @@
+package ma.norsys.bookstore.services;
+
+import ma.norsys.bookstore.exceptions.BookNotFoundException;
+import com.example.evaluation_livre.services.IBookService;
+import ma.norsys.bookstore.exceptions.PostArgumentNotValidException;
+import ma.norsys.bookstore.models.Book;
+import ma.norsys.bookstore.repositories.BookRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class BookServiceImp implements IBookService {
+
+
+    private final BookRepository bookRepository;
+
+    public BookServiceImp(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    public Book getById(int id) {
+        return bookRepository.findById(id)
+                .orElseThrow(() -> new BookNotFoundException("Not found Book with id :" + id));
+    }
+
+    @Override
+    public List<Book> getAll() {
+        return bookRepository.findAll();
+    }
+
+    @Override
+    public Book addBook(Book book) throws PostArgumentNotValidException {
+        Objects.requireNonNull(book, "Book should not be null");
+        Objects.requireNonNull(book.getTitle(), "Book name should not be null");
+        Objects.requireNonNull(book.getIsbn(), "Book Isbn should not be null");
+        Objects.requireNonNull(book.getCategories(), "Book categories should not be null");
+        Objects.requireNonNull(book.getAuthor(), "Book Author should not be null");
+        return bookRepository.save(book);
+    }
+
+    @Override
+    public boolean deleteBook(int id) {
```

Comment:
```suggestion
    public void deleteBook(int id) {
```

no need for a return